web/api/v1/api.go

```diff
@@ -602,6 +607,43 @@ type TargetDiscovery struct {
 	DroppedTargets []*DroppedTarget `json:"droppedTargets"`
 }
 
+// GlobalURLOptions contains fields used for deriving the global URL for local targets.
+type GlobalURLOptions struct {
+	ListenAddress string
+	Host          string
+	Scheme        string
+}
+
+func getGlobalURL(u *url.URL, opts GlobalURLOptions) (*url.URL, error) {
+	host, port, err := net.SplitHostPort(u.Host)
+	if err != nil {
+		return u, err
+	}
+
+	for _, lhr := range []string{"127.0.0.1", "localhost"} {
```

Comment:
FWIW this was adapted from https://github.com/prometheus/prometheus/blob/40dd13b07420a044cc1b0ca57f639c572583d9c1/web/web.go#L976, maybe it'd be good to share more code here, but I didn't press for that because we assume that the old UI (along with its code) is going away anyway.

Reply:
(if we only end up keeping the new version of this code though, it would be good to keep the comments from the original)